src/main/java/org/tinyradius/util/RadiusServer.java

```diff
@@ -353,15 +372,15 @@ protected void listen(DatagramSocket s) {
 				// check client
 				InetSocketAddress localAddress = (InetSocketAddress) s.getLocalSocketAddress();
 				InetSocketAddress remoteAddress = new InetSocketAddress(packetIn.getAddress(), packetIn.getPort());
-				String secret = getSharedSecret(remoteAddress);
+				String secret = getSharedSecret(remoteAddress, makeRadiusPacket(packetIn, "1234567890", RadiusPacket.RESERVED));
```

Comment:
It can be any string not null or empty because of this in RadiusPacket 

             // check shared secret
                if (sharedSecret == null || sharedSecret.length() == 0)
                        throw new RuntimeException("no shared secret has been set");